budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -194,43 +194,33 @@ def setUp(self):
         """Create 30 Provincial Infrastructure Projects"""
 
         self.fin_year = FinancialYear.objects.create(slug="2030-31")
-        self.url = reverse("search-provincial-infrastructure-projects")
-        self.province = u"Eastern Cape"
-        self.status = u"Construction"
-        self.source = u"Community Library Service Grant"
-        self.department = u"Department 1"
-        self.name = u"Project 1"
-        self.municipality = u"Local 2"
-        self.contractor = u"Contractor 3"
-
-        self.provinces = [
-            "Eastern Cape",
-            "Free State",
-            "Gauteng",
-            "KwaZulu-Natal",
-            "Limpopo",
-            "Mpumalanga",
-            "North West",
-            "Northern Cape",
-            "Western Cape",
-        ]
-        self.statuses = ["Design", "Tender", "Feasibility", "Construction"]
+        self.url = reverse("provincial-infrastructure-project-api")
+        self.provinces = ["Eastern Cape", "Free State"]
+        self.statuses = ["Design", "Construction"]
         self.sources = [
-            "Equitable Share",
             "Education Infrastructure Grant",
             "Community Library Service Grant",
         ]
         for i in range(30):
+            if i < 15:
```

Comment:
you're having to write logic like this because you're trying to use the same data for different tests. this is alright for now but it would be better to have one tests with lots of projects just to check paging, and the rest of the tests only need 1-3 projects, right?

Reply:
yes, I think it would be better to create many projects for pagination and have a few projects for the rest of tests. I'll keep in mind.